dvc/repo/metrics/show.py

```diff
@@ -66,13 +66,81 @@ def _read_typed_metric(typ, xpath, fd):
     return ret
 
 
+def _format_csv(content, delimiter):
+    """Format delimited text to have same column width.
+
+    Args:
+        content (str): The content of a metric.
+        delimiter (str): Value separator
+
+    Returns:
+        str: Formatted content.
+
+    Example:
+
+        >>> content = (
+            "value_mse,deviation_mse,data_set\n"
+            "0.421601,0.173461,train\n"
+            "0.67528,0.289545,testing\n"
+            "0.671502,0.297848,validation\n"
+        )
+        >>> _format_csv(content, ",")
+
+        "value_mse  deviation_mse   data_set\n"
+        "0.421601   0.173461        train\n"
+        "0.67528    0.289545        testing\n"
+        "0.671502   0.297848        validation\n"
+
+    """
+    import csv
+
+    delimiter = builtin_str(delimiter)
+    content = content.encode("UTF-8").splitlines(True)
```

Comment:
why not SrtringIO after all? It looks like it would handle new lines inside items very well.

Reply:
@shcheklein , could you explain a little bit more how would you handle this with `StringIO`?
I haven't used that module before and from the documentation, it looks like its purpose is to handle strings in a way similar to a file object.
I'm assuming that you are thinking about replacing `content.splitlines` with a `StringIO object` to pass it to the `csv.reader`, however, StringIO doesn't call `readlines` when you use the object as an iterator.

If you use `buffer.getvalue()` it converts it to a unicode, and it doesn't have a `splitlines` method :thinking:

Reply:
Sure! A pretty reasonable explanation you can find here: https://stackoverflow.com/questions/21843693/creating-stream-to-iterate-over-from-string-in-python. It implements a line by line iterator. So it should look like a file descriptor when it's used in `csv.reader`.

Reply:
Check something like this:

```python
from dvc.utils.compat import StringIO
for line in csv.reader(StringIO('x,y\n"multi\nline\nstring","single line"')):
    print(line)
```

Reply:
If we don't want to deal with newline (that can potentially break the nice output), we should at lease replace them with a single space before we print them. You can apply `map` with a replace function probably to list of items.